frc_characterization/drive_characterization/data_logger.py

```diff
@@ -311,6 +324,32 @@ def runTest(self, name, initial_speed, ramp, finished, rotate=None):
                 )
                 return
 
+            # wait for robot to say it is disabled
+            with self.lock:
+                self.lock.wait_for(lambda: self.mode == "disabled")
+
+            # tries to retrieve disabled data
+            starttime = time.time()
+            while not self.data and time.time() - starttime < timeout:
+                # print("trying to recieve data again")
+                NetworkTables.flush()
+                time.sleep(0.1)
+
+            if not self.data:
+                self.STATE.postTask(
+                    lambda: messagebox.showerror(
+                        "Timed out while trying to recieve NT data",
+                        "Maybe try running the test again?",
+                        parent=self.STATE.mainGUI,
+                    )
+                )
+                return
+
+            # cleans up data
+            data = self.data[0].split(", ")
+            data = np.array([float(s) for s in data[0:-1]])
+            data = (np.reshape(data, (-1, num_columns))).tolist()
```

Comment:
Why is it "cleaned up" this way?

Reply:
I don’t think “clean up” is the right term; I’d use “deserialize.”

Anyway, you’re right that this needs a comment explaining the data format (if you read the robot code template you can see that it’s entirely flat and comma delimited.)